### PR TITLE
Combining Repository Validating Interceptor and IG installments

### PR DESCRIPTION
**Pull request by jvitrifork**
@jamesagnew I would like your comments on this way of wiring the interceptors. Is different from the current way of doing it but it is logically the same. I would say it is a bit more type safe and stream lined, but I will only continue doing so if you're ok with it

### CODE REVIEW

**Comment by jvitrifork** [comment]
@tadgh If @jamesagnew is busy doing other stuff I would value your approval / comments on this piece

**Comment by tadgh** [comment]
Heya, I can read over this by EOD hopefully

File: src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java [hunk 1]
```diff
@@ -97,6 +97,9 @@
   @Autowired
   ApplicationContext myApplicationContext;
 
+  @Autowired(required = false)
+  RepositoryValidationInterceptorFactoryR4 factory;
+
```
Comment (tadgh) [inline src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java:102]: Does it make sense for the R4 factory to be in BaseJpaREstfulServer? Seems like those repositoryValidationInterceptorFactorys could inherit some interface, and you can autowire the interface.

File: src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java [hunk 2]
```diff
@@ -351,28 +354,24 @@ protected void initialize() throws ServletException {
     if (appProperties.getImplementationGuides() != null) {
       Map<String, AppProperties.ImplementationGuide> guides = appProperties.getImplementationGuides();
       for (Map.Entry<String, AppProperties.ImplementationGuide> guide : guides.entrySet()) {
-        packageInstallerSvc.install(new PackageInstallationSpec()
-          .setPackageUrl(guide.getValue().getUrl())
-          .setName(guide.getValue().getName())
-          .setVersion(guide.getValue().getVersion())
-          .setInstallMode(PackageInstallationSpec.InstallModeEnum.STORE_AND_INSTALL));
+			PackageInstallOutcomeJson outcome = packageInstallerSvc.install(new PackageInstallationSpec()
```
Comment (tadgh) [inline src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java:357]: Doesn't seem like this variable assignment is necessary, though I may just be missing its usage.

File: src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java [hunk 3]
```diff
@@ -351,28 +354,24 @@ protected void initialize() throws ServletException {
...
+				.setPackageUrl(guide.getValue().getUrl())
+				.setName(guide.getValue().getName())
+				.setVersion(guide.getValue().getVersion())
+				.setInstallMode(PackageInstallationSpec.InstallModeEnum.STORE_AND_INSTALL));
+
       }
     }
 
+    if(factory != null)
+    	interceptorService.registerInterceptor(factory.buildUsingStoredStructureDefinitions());
```
Comment (tadgh) [inline src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java:367]: Its just a style thing, but would appreciate braces around the if, even though its only one line.

**Review by tadgh — APPROVED** [review]
Looks ok to me with some comments.

File: src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java (hunk 1 above)
Comment (jvitrifork) [reply src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java:102]: Agree - all the leg work had been done so I've introduced the interface

File: src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java (hunk 2 above)
Comment (jvitrifork) [reply src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java]: Was used in a debug session. It shouldn't have survived

File: src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java (hunk 3 above)
Comment (jvitrifork) [reply src/main/java/ca/uhn/fhir/jpa/starter/BaseJpaRestfulServer.java]: Fixed

**Comment by jvitrifork** [comment]
@tadgh any change you could get @jamesagnew to read my last message I left him at zulip?

**Pull request merged**